frontend/src/scenes/funnels/funnelLogic.ts

```diff
@@ -342,19 +356,48 @@ export const funnelLogic = kea<funnelLogicType<TimeStepOption>>({
                 return stepsDropdown
             },
         ],
-        totalConversionRate: [
-            () => [selectors.stepsWithCount],
-            (stepsWithCount) =>
-                stepsWithCount.length > 1
-                    ? calcPercentage(stepsWithCount[stepsWithCount.length - 1].count, stepsWithCount[0].count)
-                    : 0,
-        ],
         areFiltersValid: [
             () => [selectors.filters],
             (filters) => {
                 return (filters.events?.length || 0) + (filters.actions?.length || 0) > 1
             },
         ],
+        conversionMetrics: [
+            () => [selectors.stepsWithCount, selectors.histogramStep],
+            (stepsWithCount, timeStep) => {
+                if (stepsWithCount.length <= 1) {
+                    return {
+                        average: 0,
+                        sum: 0,
+                        stepRate: 0,
+                        totalRate: 0,
+                    }
+                }
+
+                const isAllSteps = timeStep.from_step === -1
+                const fromStep = isAllSteps
+                    ? getReferenceStep(stepsWithCount, FunnelStepReference.total)
+                    : stepsWithCount[timeStep.from_step]
+                const toStep = isAllSteps ? getLastFilledStep(stepsWithCount) : stepsWithCount[timeStep.to_step]
+
+                // total sum from first step
+                // note: if last step has 0 count, it makes sense to say that total time to convert is 0
+                const sum =
+                    toStep.count === 0
+                        ? 0
+                        : stepsWithCount
+                              .slice(0, toStep.order + 1)
+                              .map((s: FunnelStep) => s.average_conversion_time * s.count) // calculate total times per step
+                              .reduce((a: number, b: number) => a + b, 0) // add it up
+
+                return {
+                    average: toStep?.average_conversion_time || 0,
```

Comment:
This seems ambiguous, `average` & `sum` could be related to time to convert or general conversion rates

Reply:
I can see that, I'll rename these to `averageTime` and `totalTime`